### DISCUSSION TITLE
Reporting - Sponsored brands for 'Product Ad' report and  Sponsored Display for 'keyword' report

### DISCUSSION THREAD

**Discussion by Vignesh-Siva96** [amzn/ads-advanced-tools-docs#165] [General]
Greetings,

I am currently using version 3 and version 2 reporting combined to get the below metrics for all ad types and aggregating them for each individual day for a particular period.

Metrics: impressions, clicks, cost, purchases7d, purchases14d, sales7d, sales14d, viewAttributedSales14d (for display ads), viewImpressions (for display ads)

Granularity: adGroupId and campaignId

I am trying to get the above metrics with the below three granularities.

1. ASIN/SKU

2. Keywords

3. Targeting

For Sponsored products, I am using 'Advertised Product' and 'Targeting' reports from version 3.

For Sponsored Brands and Sponsored Display, I am trying to use 'Product Ad' report (for ASIN/SKU and adId), 'keyword' report (for Keywords), and a 'Target' report (for targeting).

The issues I am facing are as follows,

1. The 'Product Ad' report is not available for Sponsored brands.
2. The 'keyword' report is not available for Sponsored Display.

Is there a way to overcome the above-mentioned issue?

Note: Currently I am trying to get separate reports for each granularity and aggregating them separately as well for a particular date (for each day, We will have three different reports for the same metrics based on each granularity). I would also like to know if there is a way to get all these in a single report as well.

**Comment by elizafarley** [comment]
Hi @Vignesh-Siva96, thanks for your questions! 

For issue 1: You are correct that the product ad report is not available for Sponsored Brands. There is currently no ASIN-grain reporting for Sposnored Brands since there can be multiple ASINs in a single ad. You would need to track the ASINs associated to a particular adId separately (for example, using [POST /sb/v4/ads/list](https://advertising.amazon.com/API/docs/en-us/sponsored-brands/3-0/openapi/prod#tag/Ads/operation/ListSponsoredBrandsAds)) then join to the reporting data using adId. To view ASINs purchased as a result of an ad click or view, you can also look at the [purchased product report](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/purchased-product). 

For issue 2: You are correct that there is no keyword report for Sponsored Display. Keyword is not currently a supported targeting type for Sponsored Display, which is why there is no report for keywords. For Sponsored Display, the targeting report should contain details for all associated targets.

Using the version 2 reporting APIs, you do need to make separate calls for each level of granularity and date. In [version 3 reporting](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/overview) (SB and SD are newly supported), you are able to request reports that span multiple days, and keywords and targets are combined into a single report. Another option is to look into [Amazon Marketing Stream](https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/overview), which provides you with campaign, ad group, ad, and target/keyword in a single record. 

Let me know if you have any other questions.

**Comment by Vignesh-Siva96** [comment]
Hi Eliza,

Thanks for the clear response.

Also, can I fetch the Product name based on the ASIN from the ads API?

**Reply by elizafarley** [reply]
Yes, you should be able to get product name from ASIN using the [product metadata API](https://advertising.amazon.com/API/docs/en-us/product-metadata). However, it will only work for ASINs tied to your Amazon advertising account.
